### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: projectwriter.py [hunk 1]
```diff
@@ -834,9 +835,11 @@ def __get_osl_node_params(self, node, scene):
         return parameters
 
     def __get_socket_links(self, node):
+        connections = []
         for output in node.outputs:
             if output.is_linked:
-                connections = [node.name, output.socket_osl_id, output.links[0].to_node.name, output.links[0].to_socket.socket_osl_id]
+                for x in range(0, len(output.links)):
```
Comment (dictoon) [inline projectwriter.py]: Why not

    for link in output.links:
        ...

?

Comment (jdent02) [reply projectwriter.py]: Good question. Using that terminology, ‘link’ is actually a full NodeItem reference, which can’t be used as a list index in the following argument.  Took me some thinking to get around that.

Comment (dictoon) [reply projectwriter.py]: > which can’t be used as a list index in the following argument

But then you don't use it as a list index, you use it directly.

Comment (jdent02) [reply projectwriter.py]: I can alter it if needed.

Comment (dictoon) [reply projectwriter.py]: Sure, just smuggle that into a future PR :)

**Pull request merged**